src/aspire/denoising/__init__.py

```diff
@@ -0,0 +1,50 @@
+import logging
+import numpy as np
+
+from aspire.source import ArrayImageSource
+from aspire.image import Image
+from aspire.io.starfile import save_star
+
+logger = logging.getLogger(__name__)
+
+
+class Denoiser:
+    """
+    Define a base class for denoising 2D images
+    """
+    def __init__(self, src):
+        """
+        Initialize an object for denoising 2D images from the image source
+
+        :param src: The source object of 2D images with metadata
+        """
+        self.src = src
+
+        self.nimg = src.n
+
+    def denoise(self):
+        """
+        Denoise 2D images
+        """
+        raise NotImplementedError('subclasses must implement this')
+
+    def src_denoised(self):
+        """
+        Return an ImageSource object with denoised images
+        """
+        src = ArrayImageSource(Image(self.imgs_estim))
```

Comment:
Ideally, this would be performed by the subclass since we don't always want to store all the images in memory. Could we have this method be abstract (just like `denoise` above) and let the subclass implement it?

Also, maybe `to_src` is a better name? Or just `src`? It should already be implied that they're denoised images.

Reply:
Renamed to `to_src` and made it to abstract in commit  f8d647c.